sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DefaultDocumentQueryExecutionContext.java

```diff
@@ -186,7 +186,13 @@ protected PartitionKeyInternal getPartitionKeyInternal() {
                                                     this.fetchExecutionRangeAccumulator.getExecutionRanges(),
                                                     Arrays.asList(schedulingTimeSpanMap)),
                                             tFeedResponse.getActivityId());
-                            BridgeInternal.putQueryMetricsIntoMap(tFeedResponse, DEFAULT_PARTITION_RANGE, qm);
+                            String pkrId = tFeedResponse.getResponseHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
```

Comment:
why can't we rely on the targetRange please see here:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java#L93

is that going be different than this one from the header response?

Reply:
We need both feedrange and rangeId in query metric key , it will be easier to diagnose issue for On call, discussed with @mbhaskar  please check this comment https://github.com/Azure/azure-sdk-for-java/pull/22202#discussion_r655717122